tst/com/amazon/corretto/crypto/provider/test/HmacTest.java

```diff
@@ -66,10 +74,75 @@ public class HmacTest {
     }
   }
 
+  // TODO: This version needs to be updated once known.
+  private static final String MINIMUM_VERSION_WITH_PRECOMPUTED_KEY = "2.4.1";
+
+  /**
+   * Call this function before any tests that use precomputed keys.
+   *
+   * <p>TODO: This function will need to be changed once FIPS support precomputed keys.
+   */
+  private void assumePrecomputedKeySupport() {
+    TestUtil.assumeMinimumVersion(
+        MINIMUM_VERSION_WITH_PRECOMPUTED_KEY, AmazonCorrettoCryptoProvider.INSTANCE);
+    Assumptions.assumeFalse(
+        TestUtil.isFips(), "precomputed keys are only supported in non-FIPS builds");
+  }
+
+  /**
+   * Call this function before any tests that assume precomputed keys are not supported.
+   *
+   * <p>TODO: This function will need to be changed once FIPS support precomputed keys.
+   */
+  private void assumeNoPrecomputedKeySupport() {
+    Assumptions.assumeTrue(TestUtil.isFips());
+  }
+
   private static List<String> supportedHmacs() {
     return SUPPORTED_HMACS;
   }
 
+  private int getPrecomputedKeyLength(String algorithm) {
+    int precomputedKeySize;
+    switch (algorithm) {
+      case "HmacMD5":
+        precomputedKeySize = 16;
```

Comment:
nit: suggest adding a comment explaining these values. are these just the digest size for each MAC?

Reply:
done